lstchain/scripts/lstchain_dl1ab.py

```diff
@@ -142,19 +159,21 @@ def main():
                     dl1_container.width = width
                     dl1_container.length = length
                     dl1_container.r = np.sqrt(dl1_container.x ** 2 + dl1_container.y ** 2)
-
-                else:
-                    # for consistency with r0_to_dl1.py:
-                    for key in dl1_container.keys():
-                        dl1_container[key] = \
-                            u.Quantity(0, dl1_container.fields[key].unit)
-
-                    dl1_container.width = u.Quantity(np.nan, u.m)
-                    dl1_container.length = u.Quantity(np.nan, u.m)
-                    dl1_container.wl  = u.Quantity(np.nan, u.m)
-
-            for p in parameters_to_update:
-                params[ii][p] = u.Quantity(dl1_container[p]).value
+                    dl1_container.log_intensity = np.log10(dl1_container.intensity)
+
+                disp_dx, disp_dy, disp_norm, disp_angle, disp_sign = disp(dl1_container['x'].to_value(u.m),
+                                                                          dl1_container['y'].to_value(u.m),
+                                                                          params['src_x'][ii],
+                                                                          params['src_y'][ii],
+                                                                          )
+                dl1_container['disp_dx'] = disp_dx
+                dl1_container['disp_dy'] = disp_dy
+                dl1_container['disp_norm'] = disp_norm
+                dl1_container['disp_angle'] = disp_angle
+                dl1_container['disp_sign'] = disp_sign
```

Comment:
These definitions are not indented within `if n_pixels > 0` condition. Should they be or it does not make any difference?

Reply:
actually, I had to put them here to recompute it even when `n_pixels=0` and in this case they are equal to `nan`

Reply:
I see, right